### ISSUE TITLE
Can't login ccxt.base.errors.ExchangeError: ftx {"error":"Not logged in","success":false}

### ISSUE THREAD

**Issue by unheaven** [ccxt/ccxt#7554]
I think I put API key and secret it right.
Can't login ccxt.base.errors.ExchangeError: ftx {"error":"Not logged in","success":false}
- FTX
- OS: windows 10
- Programming Language version: Python version: 3.7
- CCXT version:1.33.90

```
REPLACE_THIS_WITH_YOUR_CODE_TO_REPRODUCE_THE_ISSUE_WITHOUT_YOUR_KEYS
```
import ccxt
print(ccxt.exchanges)

ftx = ccxt.ftx({
    'apiKey': "key",
    'secret': "secret",
    'enableRateLimit': True,
})
print(ftx.fetch_balance())
```
REPLACE_THIS_WITH_YOUR_OUTPUT_ERROR_EXCEPTION_IN_TEXT_NO_SCREENSHOTS
```
C:\Users\UnHeaven\Desktop\FTXBOT\venv\Scripts\python.exe C:/Users/UnHeaven/Desktop/FTXBOT/XRPBULL.py
['acx', 'aofex', 'bcex', 'bequant', 'bibox', 'bigone', 'binance', 'binanceje', 'binanceus', 'bit2c', 'bitbank', 'bitbay', 'bitfinex', 'bitfinex2', 'bitflyer', 'bitforex', 'bitget', 'bithumb', 'bitkk', 'bitmart', 'bitmax', 'bitmex', 'bitpanda', 'bitso', 'bitstamp', 'bitstamp1', 'bittrex', 'bitvavo', 'bitz', 'bl3p', 'bleutrade', 'braziliex', 'btcalpha', 'btcbox', 'btcmarkets', 'btctradeua', 'btcturk', 'buda', 'bw', 'bybit', 'bytetrade', 'cex', 'chilebit', 'coinbase', 'coinbaseprime', 'coinbasepro', 'coincheck', 'coinegg', 'coinex', 'coinfalcon', 'coinfloor', 'coingi', 'coinmarketcap', 'coinmate', 'coinone', 'coinspot', 'coss', 'crex24', 'currencycom', 'deribit', 'digifinex', 'dsx', 'eterbase', 'exmo', 'exx', 'fcoin', 'fcoinjp', 'flowbtc', 'foxbit', 'ftx', 'gateio', 'gemini', 'hbtc', 'hitbtc', 'hollaex', 'huobijp', 'huobipro', 'huobiru', 'ice3x', 'idex', 'independentreserve', 'indodax', 'itbit', 'kraken', 'kucoin', 'kuna', 'lakebtc', 'latoken', 'lbank', 'liquid', 'livecoin', 'luno', 'lykke', 'mercado', 'mixcoins', 'oceanex', 'okcoin', 'okex', 'paymium', 'phemex', 'poloniex', 'probit', 'qtrade', 'rightbtc', 'southxchange', 'stex', 'stronghold', 'surbitcoin', 'therock', 'tidebit', 'tidex', 'timex', 'upbit', 'vaultoro', 'vbtc', 'wavesexchange', 'whitebit', 'xbtce', 'xena', 'yobit', 'zaif', 'zb']
Traceback (most recent call last):
  File "C:\Users\UnHeaven\Desktop\FTXBOT\venv\lib\site-packages\ccxt\base\exchange.py", line 585, in fetch
    response.raise_for_status()
  File "C:\Users\UnHeaven\Desktop\FTXBOT\venv\lib\site-packages\requests\models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: UNAUTHORIZED for url: https://ftx.com/api/wallet/balances

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/Users/UnHeaven/Desktop/FTXBOT/XRPBULL.py", line 9, in <module>
    print(ftx.fetch_balance())
  File "C:\Users\UnHeaven\Desktop\FTXBOT\venv\lib\site-packages\ccxt\ftx.py", line 789, in fetch_balance
    response = self.privateGetWalletBalances(params)
  File "C:\Users\UnHeaven\Desktop\FTXBOT\venv\lib\site-packages\ccxt\base\exchange.py", line 465, in inner
    return entry(_self, **inner_kwargs)
  File "C:\Users\UnHeaven\Desktop\FTXBOT\venv\lib\site-packages\ccxt\base\exchange.py", line 490, in request
    return self.fetch2(path, api, method, params, headers, body)
  File "C:\Users\UnHeaven\Desktop\FTXBOT\venv\lib\site-packages\ccxt\base\exchange.py", line 486, in fetch2
    return self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File "C:\Users\UnHeaven\Desktop\FTXBOT\venv\lib\site-packages\ccxt\base\exchange.py", line 597, in fetch
    self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
  File "C:\Users\UnHeaven\Desktop\FTXBOT\venv\lib\site-packages\ccxt\ftx.py", line 1472, in handle_errors
    raise ExchangeError(feedback)  # unknown message
ccxt.base.errors.ExchangeError: ftx {"error":"Not logged in","success":false}

**Comment by kroitor** [comment]
TL;DR: make sure you use the proper keys and your system clock is in synch with the exchange.

Duplicate:

- https://github.com/ccxt/ccxt/issues/6985
- https://github.com/ccxt/ccxt/issues/7244
- https://github.com/ccxt/ccxt/issues/7461
- https://github.com/ccxt/ccxt/search?q=ftx+not+logged+in&type=Issues
- https://blog.ftx.com/blog/api-authentication/#not-logged-in-error

For the future:

- https://github.com/ccxt/ccxt/wiki/FAQ#what-is-required-to-get-help

Let us know if that does not help.

**Comment by kalamengde** [comment]
sub account  api log in  fail , 


exchange = ccxt.ftx ({
  "enableRateLimit": "true",
  "apiKey": "xxxx",
  "secret": "xxxxxxa",
  "headers": {'FTX-SUBACCOUNT': "my_subaccount_cake"},
})
print (exchange)

ftx_set_order=exchange.private_post_orders({'market': 'CAKE-perp', 'side': 'buy',  'price': 11,"size":"1"})
